go-app-ussd.js

```diff
@@ -66,12 +66,17 @@ go.app = function() {
         });
 
         self.states.add('states:select_language', function(name){
+            var language_previously_not_set = self.im.user.lang === null;
```

Comment:
This check is okay, but I don't think this state will work because you're returning the `LanguageChoice` state regardless of whether `lang` is `null`. I think likely you want to delegate to the main menu if `lang` is not `null`?